Fix/separation on steps and floor

emmanuelOpenForge opened the pull request:
# Pull request type

-   [x] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] Documentation content changes
-   [ ] E2E Test(s)
-   [ ] Other (please describe):

# What is the current behavior?
Issues with floor and steps, and obstacles
# What is the new behavior?
- Removed general gravity everything is going to have their own one
- Added gravity to obstacles, character and poop of pigeons
- Now the new floors are instantiated as sprites and afterwards they become to tileSprites this help us to avoid going through them
- Added listeners for left and right keyboard to set properly the flags
- Fixed pigeons to spawn from the floor and not being dropped
- Fixed initial position of the poop
- Fixed velocity when going up of the stairs
# Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

# Other information

https://github.com/openforge/rock-the-steps-app/assets/106994064/2eb72c47-4f2f-4593-bda6-32ed09164319

emmanuel-Andela committed:
Added logic display objects in the floor not being droped
- Added logic display objects in the floor depending on the size of the floor so they will appear from the correct level

emmanuel-Andela committed:
Merge branch 'develop' into fix/SeparationOnStepsAndFloor

Ricardo385 commented on libs/shared/phaser-singleton/src/lib/scenes/world.scene.ts:
```diff
@@ -142,12 +142,22 @@ export class WorldScene extends Phaser.Scene {
         this.character = new Character(this, this.firstFloor.sprite);
         this.physics.add.existing(this.firstFloor.sprite, true);
         this.pointsText = this.add.text(INITIAL_POINTS_X, INITIAL_POINTS_Y, '0', { fontSize: '3vh', color: 'black' });
+
```
Should we remove this code commented?

Ricardo385 reviewed (approved):
approved but left a minor comment

emmanuel-Andela committed:
Added fix for disconnection for stairs and floor trough
- Removed general gravity everything is going to have their own one
- Added gravity to obstacles, character and poop of pigeons
- Now the new floors are instantiated as sprites and afterwards they become to tileSprites this help us to avoid going through them
- Added listeners for left and right keyboard to set properly the flags
- Fixed pigeons to spawn from the floor and not being dropped
- Fixed initial position of the poop
- Fixed velocity when going up of the stairs

emmanuel-Andela committed:
Added fix for disconnection for stairs and floor trough
- Removed general gravity everything is going to have their own one
- Added gravity to obstacles, character and poop of pigeons
- Now the new floors are instantiated as sprites and afterwards they become to tileSprites this help us to avoid going through them
- Added listeners for left and right keyboard to set properly the flags
- Fixed pigeons to spawn from the floor and not being dropped
- Fixed initial position of the poop
- Fixed velocity when going up of the stairs

emmanuel-Andela committed:
Merge branch 'develop' into fix/SeparationOnStepsAndFloor

# Conflicts:
#	libs/shared/phaser-singleton/src/lib/scenes/world.scene.ts
#	libs/shared/phaser-singleton/src/lib/utilities/hud-helper.ts
